debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/parser/LogMinerDmlParser.java

```diff
@@ -112,9 +117,13 @@ private LogMinerDmlEntry parseInsert(String sql) {
                 throw new DmlParserException("Columns: " + columnNames + ", Values: " + columnValues);
             }
 
-            List<LogMinerColumnValue> newValues = new ArrayList<>(columnNames.size());
-            for (int i = 0; i < columnNames.size(); ++i) {
-                newValues.add(createColumnValue(columnNames.get(i), columnValues.get(i)));
+            final Map<String, String> columnMap = createMap(columnNames, columnValues);
```

Comment:
Why do we need this map? The original index-based look-up seemed much more efficient?

Reply:
The index-based look-up was strictly based on the order the columns/values appeared in the generated SQL.  Since there is no where in the documentation that states that Oracle LogMiner will generate the SQL with columns in table order, I added this to guarantee that the columns are emitted in the order the table schema dictates.  We _might_ be able to get away without doing this for inserts, but we'd need to test this to be absolutely certain.

Reply:
But aren't column names and values part of the statement itself? I.e. why do need to reach back to the `Table` metadata to begin with?

Reply:
The column names and values are provided, but there is no guarantee that the order in which the generated SQL specifies the tuples are in the same column order as the `Table`'s metadata.  Couldn't this create an issue when we're taking the `Object[]` new/old column values and emitting the event with the appropriate converter?

Reply:
Can't we look up the right `Column` though based on the name of the incoming column value/name tuple?

Reply:
Perhaps I'm not following, so lets assume we have a table defined as:
```
CREATE TABLE t (ID numeric primary key, DATA1 VARCHAR2(50), DATA2 DATE);
```
The `INSERT` in the LogMiner result-set comes in as follows:
```
INSERT INTO t (ID, DATA2, DATA1) VALUES (1, TO_DATE('2021-03-19'), 'Hello World');
```
The parser simply traverses the SQL statement and creates 3 pairs in the order they're observed by name:
* ID <-> 1
* DATA2 <-> TO_DATE('2021-03-19')
* DATA1 <-> 'Hello World'

My understanding is this tuple should actually be ordered as follows to adhere to the table schema definition so that when we convert the pairs into the value object array, they line up with the converters like as follows:
* ID <-> 1
* DATA1 <-> 'Hello World'
* DATA2 <-> TO_DATE('2021-03-19')

This is why we tie back into the `Table` metadata.  If this isn't required I can certainly remove this step but that was at least my interpretation of how the value object arrays are being used for new/old values in the change emitter logic.

Reply:
Based on our discussions, we'll leave it as-is for now and if it becomes an issue we'll circle back.